### PR TITLE
Applying hass language attribute changes

### PR DESCRIPTION
**Pull request by JonahKr**
This is addressing the Changes from https://github.com/home-assistant/frontend/pull/7925
which basically moved the `hass.language` attribute to  `hass.locale.language` and adds `hass.locale.number_format`.
Considering the impact on past versions (below v20210401) there should be an additional layer for supporting  both formats at least for the foreseeable future.

### CODE REVIEW

File: dist/types.d.ts [hunk 1]
```diff
@@ -159,7 +159,11 @@ export interface HomeAssistant {
     selectedTheme?: string | null;
     panels: Panels;
     panelUrl: string;
+    /**
+    * @deprecated Use `locale.language` instead
+    */
```
Comment (JonahKr) [inline dist/types.d.ts:164]: I am not sure if this is necessary or if this would even be displayed for users.

**Comment by JonahKr** [comment]
One Issue I see with this is backwards compatibility.
Older hass-instances (<20200401)  don't have the `locale` attribute thus won't work with this upgrade.

**Comment by iantrich** [comment]
I'll have to give the backwards compatibility part a thought. I think documenting the minimum version for the module version should be fine but if we can think of how to do it both was it would be better.

**Comment by JonahKr** [comment]
I added a `extract_locale` function which basically uses the hass object to create a valid locale attribute structure.
Don't know if that would be the fitting solution though.
It would need to be implemented in every usecase of locale in cch or the users has to use it as they prefer.

**Comment by JonahKr** [comment]
I just had a different idea. Maybe we could just let the user pass the hass object and unpack it with the `extract_locale` in the `format_number` function itsself

**Comment by JonahKr** [comment]
@iantrich any update?

**Comment by marksie1988** [comment]
Would be good to get this implemented, any update on this?

**Comment by JonahKr** [comment]
@marksie1988 so this PR atm only implements it for format number, but the procedure would be fairly similar for the date one added in the last version.
One open point until now was about how to handle backwards compatibility.
Maybe you have an idea regarding this?

**Comment by iantrich** [comment]
I think we can probably just move forward now. I'll take a look soon.

**Comment by JonahKr** [comment]
if we ignore backwards compatability, i would have to remove the `extract locale` function

**Comment by marksie1988** [comment]
@JonahKr I personally don't think back compatibility is needed, that's the reason for versioning. 

Backwards compatibility can get messy and complicate usage. 

For me I can just add a table to say vX.X works with HA version x and before. Etc.

**Comment by iantrich** [comment]
Yeah that sounds fine.

**Comment by JonahKr** [comment]
@iantrich something wrecked the docs in my fork. can you run the docs command after merging?

**Comment by iantrich** [comment]
Published: https://www.npmjs.com/package/custom-card-helpers/v/1.7.1

**Pull request merged**